tel2puml/sequence_otel_v2.py

```diff
@@ -257,3 +255,54 @@ def sequence_otel_jobs(
         yield sequence_otel_event_job(
             job, async_flag, event_to_async_group_map
         )
+
+
+def sequence_otel_job_id_streams(
+    job_id_streams: Generator[Generator[OTelEvent, Any, None], Any, None],
+    async_flag: bool = False,
+    event_to_async_group_map: dict[str, dict[str, str]] | None = None,
+) -> Generator[Generator[PVEvent, Any, None], Any, None]:
+    """
+    Sequence OTel events in multiple jobs.
+
+    :param job_id_streams: A generator of generators, where each inner
+    generator yields OTelEvent objects grouped by job ID.
+    :type job_id_streams: `Generator`[`Generator`[:class:`OTelEvent`, `Any`,
+    `None`], `Any`, `None`]
+    :param async_flag: A flag indicating whether to sequence event groups
+    asynchronously or not, defaults to False.
+    :type async_flag: `bool`
+    :param event_to_async_group_map: A dictionary mapping event types to
+    groups of events that occur asynchronously, defaults to None.
+    :type event_to_async_group_map: `dict`[`str`, `dict`[`str`, `str`]] |
+    `None`
+    :return: A generator of jobs (generators) of PVEvents.
+    :rtype: `Generator`[`Generator`[:class:`PVEvent`, `Any`, `None`], `Any`,
+    `None`]
+    """
+
+    def job_ids_to_eventid_to_otelevent_map(
```

Comment:
shouldn't be nested in another function. just extract out and then test this separately

Reply:
separated it out